sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1733,4 +1792,43 @@ static uint32_t _handle_rio(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
 
     return route_ltime;
 }
+
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_QUEUE_PKT)
+gnrc_pktqueue_t *_nbr_pop_pkt(_nib_onl_entry_t *node)
+{
+    if (node->pktqueue_len == 0) {
+        assert(node->pktqueue == NULL);
+        return NULL;
+    }
+    assert(node->pktqueue != NULL);
+
+    node->pktqueue_len--;
+
+    return gnrc_pktqueue_remove_head(&node->pktqueue);
+}
+
+void _nbr_push_pkt(_nib_onl_entry_t *node, gnrc_pktqueue_t *pkt)
+{
+    assert(_get_nud_state(node) == GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE);
+
+    if (ARRAY_SIZE(_queue_pool) > UINT8_MAX && node->pktqueue_len == UINT8_MAX) {
```

Comment:
I would re-introduce a general cap here (say 16), out of following reasons:
 1. less probable to deplete the cache, thus less probable the need to go through all neighbors to drop packets
 2. queueing on a neighbor is O(nbr packet queue len)
 3. https://www.rfc-editor.org/rfc/rfc4861#section-7.2.2 clearly recomends this: *The queue MUST hold at least one packet, and MAY contain more.  However, the number of queued packets per neighbor SHOULD be limited to some small value.*

Reply:
Good idea, if more than a few packets are queued something is fishy already, we don't need to find the greatest hog, a hog above a certain threshold would already suffice

Reply:
There's a cap now.